### PR TITLE
using online file instead to fix NUMA nodes init

### LINKED ISSUE #83175: kubelet service failed with wrong number of possible NUMA nodes
**What happened**: With CentOS 7.6, the kubelet systemd service may failed, because the numbers of NUMA nodes it reports is different from the actual number of nodes. the kubelet log is:

Sep 25 10:26:37 node50 systemd: Started Kubernetes systemd probe.
Sep 25 10:26:37 node50 kubelet: I0925 10:26:37.696734  275300 server.go:410] Version: v1.16.0
Sep 25 10:26:37 node50 kubelet: I0925 10:26:37.697108  275300 plugins.go:100] No cloud provider specified.
Sep 25 10:26:37 node50 kubelet: I0925 10:26:37.697143  275300 server.go:773] Client rotation is on, will bootstrap in background
Sep 25 10:26:37 node50 kubelet: I0925 10:26:37.719287  275300 certificate_store.go:129] Loading cert/key pair from "/var/lib/kubelet/pki/kubelet-client-current.pem".
Sep 25 10:26:37 node50 kubelet: I0925 10:26:37.962954  275300 server.go:644] --cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /
Sep 25 10:26:37 node50 kubelet: F0925 10:26:37.963578  275300 server.go:271] failed to run Kubelet: open /sys/devices/system/node/node2/cpulist: no such file or directory
Sep 25 10:26:37 node50 systemd: kubelet.service: main process exited, code=exited, status=255/n/a
Sep 25 10:26:37 node50 systemd: Unit kubelet.service entered failed state.

**What you expected to happen**: 
But this node has only 2 NUMA nodes, node0 and node1
[root@node50 ~]# lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                72
On-line CPU(s) list:   0-71
Thread(s) per core:    2
Core(s) per socket:    18
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Gold 6240Y CPU @ 2.60GHz
Stepping:              7
CPU MHz:               1000.073
CPU max MHz:           3900.0000
CPU min MHz:           1000.0000
BogoMIPS:              5200.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              25344K
NUMA node0 CPU(s):     0-17,36-53
NUMA node1 CPU(s):     18-35,54-71
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 ......

**How to reproduce it (as minimally and precisely as possible)**: 
With Intel Xeon CLX-SP 2 sockets CPUs, in CentOS 7.6, sometimes the sysfs /sys/devices/system/node/possible reports more than actual CPU counts. And the topology in cpumanager read this file, if it reads 4 CPUs, but there are only 2 CPUs, then kubelet service will failed.

[root@node50 ~]# cat /sys/devices/system/node/possible
0-3
[root@node50 ~]# cat /sys/devices/system/node/online
0-1

**Anything else we need to know?**:
Suggest func GetNUMANodeInfo() read /sys/devices/system/node/online instead of /sys/devices/system/node/possible

**Environment**:
- Kubernetes version (use `kubectl version`): 
[root@node50 ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: 
2x socket Intel Xeon Gold 6240Y
- OS (e.g: `cat /etc/os-release`): 
[root@node50 ~]# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

- Kernel (e.g. `uname -a`):
[root@node50 ~]# uname -a
Linux node50 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by zouyee**
**What type of PR is this?**
 /kind bug
/kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #83175

**Special notes for your reviewer**:

using /sys/devices/system/node/online instead of /sys/devices/system/node/possible

**Does this PR introduce a user-facing change?**:

```release-note
Use online nodes instead of possible nodes when discovering available NUMA nodes
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by odinuge — APPROVED** [review]
Good catch! I think the comment should be updated too, to make it clear why we should use online and not possible.


I guess this problem occurs when a machine has more cpu sockets than it has cpus installed (eg.  1 socket with a cpu, and one empty one).

Ref. the `TODO: This is a temporary workaround until cadvisor provides this` we should probably think about this when implementing the discovery in cadvisor too. 

/lgtm
/priority important-soon
/cc klueska ConnorDoyle lmdaly
ref. https://github.com/kubernetes/kubernetes/issues/49964 https://github.com/kubernetes/community/blob/cd961fad1e9d1c2c2ddf85e36a2f93c202c1da89/contributors/design-proposals/node/topology-manager.md

File: pkg/kubelet/cm/cpumanager/topology/topology.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func GetNUMANodeInfo() (NUMANodeInfo, error) {
 	// nil NUMANodeInfo, indicating that no NUMA information is available
 	// on this machine. This should implicitly be interpreted as having a
 	// single NUMA node with id 0 for all CPUs.
-	nodelist, err := ioutil.ReadFile("/sys/devices/system/node/possible")
+	nodelist, err := ioutil.ReadFile("/sys/devices/system/node/online")
```
Comment (ConnorDoyle) [inline pkg/kubelet/cm/cpumanager/topology/topology.go]: :+1:

**Review by ConnorDoyle — APPROVED** [review]
Thanks!

/approve

**Comment by odinuge** [comment]
I think this deserves a backport into the v1.16 release branch. Would you care to make that @zouyee?

**Pull request merged**